analysis/runtime/src/parse.rs

```diff
@@ -0,0 +1,97 @@
+//! Don't want a huge depedency on `clap`.
+
+use std::{
+    ffi::OsStr,
+    fmt::{self, Debug, Display, Formatter},
+    marker::PhantomData,
+};
+
+pub trait AsStr {
+    fn as_str(&self) -> &'static str;
+}
+
+pub struct Choices<T> {
+    _phantom: PhantomData<T>,
+}
+
+impl<T> Default for Choices<T> {
+    fn default() -> Self {
+        Self {
+            _phantom: Default::default(),
+        }
+    }
+}
+
+pub trait GetChoices
+where
+    Self: Sized,
+{
+    fn choices() -> &'static [Self];
+}
+
+impl<T: GetChoices + AsStr + 'static> Debug for Choices<T> {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        f.write_str("one of ")?;
+        f.debug_list()
+            .entries(T::choices().iter().map(|choice| choice.as_str()))
+            .finish()?;
+        Ok(())
+    }
+}
+
+impl<T: GetChoices + AsStr + 'static> Display for Choices<T> {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        write!(f, "{self:?}")
+    }
+}
+
+impl AsStr for bool {
+    fn as_str(&self) -> &'static str {
+        match self {
+            true => "true",
+            false => "false",
+        }
+    }
+}
+
+impl GetChoices for bool {
+    fn choices() -> &'static [Self] {
+        &[true, false]
+    }
+}
+
+pub fn one_of<T: GetChoices + AsStr + 'static>(s: &OsStr) -> Result<&'static T, String> {
+    T::choices()
+        .iter()
+        .find(|choice| s == OsStr::new(choice.as_str()))
+        .ok_or_else(|| {
+            let s = s.to_string_lossy().into_owned();
```

Comment:
I think deref should work here instead of `into_owned`, and the same for the other instances of this pattern.

Reply:
Ah yes, I didn't realize that works.  It doesn't need ownership, but I didn't know `Cow` just works nicely like that.